src/main/java/com/amazon/dlic/auth/http/saml/HTTPSamlAuthenticator.java

```diff
@@ -275,14 +269,26 @@ public Void run() throws InitializationException {
         }
     }
 
-    private AbstractReloadingMetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
+    private AbstractBatchMetadataResolver createMetadataResolver(final Settings settings, final Path configPath)
             throws Exception {
-        final AbstractReloadingMetadataResolver metadataResolver;
+        final AbstractBatchMetadataResolver metadataResolver;
+        final String idpMetadataUrl = settings.get("idp.metadata_url");
+        final String idpMetadataFile = settings.get("idp.metadata_file");
+        final String idpMetadataBody = settings.get("idp.metadata_body");
 
         if (idpMetadataUrl != null) {
-            metadataResolver = new SamlHTTPMetadataResolver(settings, configPath);
+            metadataResolver = new SamlHTTPMetadataResolver(idpMetadataUrl, settings, configPath);
+            ((SamlHTTPMetadataResolver)metadataResolver).setMinRefreshDelay(settings.getAsLong("idp.min_refresh_delay", 60L * 1000L));
+            ((SamlHTTPMetadataResolver)metadataResolver).setMaxRefreshDelay(settings.getAsLong("idp.max_refresh_delay", 14400000L));
+            ((SamlHTTPMetadataResolver)metadataResolver).setRefreshDelayFactor(settings.getAsFloat("idp.refresh_delay_factor", 0.75f));
         } else {
```

Comment:
If you initialize `idpMetadataUrl`, `idpMetadataFile` and `idpMetadataBody` outside of `if` block, use `else if`. I used `else` in the suggestion to initialize `idpMetadataFile`.